src/arrayMethodSort.js

```diff
@@ -5,7 +5,37 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let compareFunctionMain = compareFunction;
+
+    if (typeof compareFunction !== 'function') {
+      compareFunctionMain = (a, b) => {
+        const valueA = '' + a;
+        const valueB = '' + b;
+
+        if (valueA > valueB) {
+          return 1;
+        }
+
+        if (valueA < valueB) {
+          return -1;
+        }
+
+        return 0;
+      };
+    }
```

Comment:
- Move the default compare function outside the sort method and you can just assign it as a default parameter
- Don't use concatenation to create strings, use template strings, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals